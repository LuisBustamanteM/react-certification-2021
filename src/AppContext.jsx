import React, {createContext, useReducer} from 'react'

export function appReducer(state, action) {
    switch (action.type) {
        case 'UPDATE_QUERY': {
            return {
                ...state,
                query: action.value
            }
        }
        case 'TOGGLE_MODE': {
            return {
                ...state,
                darkMode: action.value
            }
        }
        case 'GET_VIDEOS': {
            return {
                ...state,
                videos: action.value
            }
        }
        case 'ADD_FAVORITES': {
            let storedIds = JSON.parse(window.sessionStorage.getItem('favoriteVideos')) || []
            storedIds = [...storedIds, action.value]
            window.sessionStorage.setItem("favoriteVideos", JSON.stringify(storedIds))

            return {
                ...state,
                favoriteIds: [
                    ...state.favoriteIds,
                    action.value
                ]
            }
        }
        case 'REMOVE_FAVORITES': {
            let storedIds = JSON.parse(window.sessionStorage.getItem('favoriteVideos')) || []
            let removedIds = storedIds.filter(id => id !== action.value)
            window.sessionStorage.setItem("favoriteVideos", JSON.stringify(removedIds))

            return {
                ...state,
                favoriteIds: removedIds
            }
        }
        case 'LOGIN': {
            window.sessionStorage.setItem("isLoggedIn", "true")
            window.sessionStorage.setItem("userData", JSON.stringify(action.value))
            return {
                ...state,
                isLoggedIn: true,
                userData: action.value
            }
        }
        case "LOGOUT": {
            window.sessionStorage.removeItem("isLoggedIn")
            window.sessionStorage.removeItem("userData")
            return {
                ...state,
                isLoggedIn: false,
                userData: {}
            }
        }
        default:
            break
    }

    return state
}

export const initialState = {
    darkMode: true,
    query: "Wizeline",
    videos: [],
    favoriteIds: JSON.parse(window.sessionStorage.getItem("favoriteVideos")) || [],
    isLoggedIn: !!window.sessionStorage.getItem("isLoggedIn"),
    userData: JSON.parse(window.sessionStorage.getItem("userData")) || {},
}

export const StateContext = createContext()
export const DispatchContext = createContext()

const AppContext = (props) => {
    const [state, dispatch] = useReducer(appReducer, initialState)

    return (
        <DispatchContext.Provider value={dispatch}>
            <StateContext.Provider value={state}>
                {props.children}
            </StateContext.Provider>
        </DispatchContext.Provider>

    )

}

export default AppContext

