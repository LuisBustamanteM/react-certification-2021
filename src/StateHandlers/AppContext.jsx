import React, {createContext, useReducer} from 'react'
import {appReducer, initialState} from "./AppReducer";

export const StateContext = createContext()
export const DispatchContext = createContext()

const AppContext = (props) => {
    const [state, dispatch] = useReducer(appReducer, initialState)

    return (
        <DispatchContext.Provider value={dispatch}>
            <StateContext.Provider value={state}>
                {props.children}
            </StateContext.Provider>
        </DispatchContext.Provider>

    )

}

export default AppContext

