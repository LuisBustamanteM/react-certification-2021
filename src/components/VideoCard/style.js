import {colors} from '../../globalStyle';
import styled from 'styled-components';

export const VideoCardContainer = styled.div`
    width: 345px;
    height: 345px;
    border: solid 0.8px #ffff;
    border-radius: 4px;
    box-shadow: 0px 1px 1px black;
    overflow: hidden;
    margin: 9px;
`;

export const Thumbnail = styled.div`
    display: block;
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center;
    background-image: url(${props => props.highImage});
    height: 140px;
    @media screen and (max-width: 1200px) {
        background-image: url(${props => props.mediumImage});
    }
    @media screen and (max-width: 800px) {
        background-image: url(${props => props.defaultImage});
    }
`;

export const Content = styled.div`
    padding: 15px
`;

export const Title = styled.h2`
    font-size: 1em;
`;

export const Paragraph = styled.p`
    font-size: 0.8em;
    color: ${colors.dimText};
`;