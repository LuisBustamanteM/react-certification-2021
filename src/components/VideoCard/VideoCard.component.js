import React from 'react';
import {VideoCardContainer, Thumbnail, Title, Paragraph, Content} from './style';
import {Link} from 'react-router-dom'

const VideoCardComponent = (props) => {
    return (
        <VideoCardContainer>
            <Link to={`/video/${props.id}`}>
                <Thumbnail defaultImage={props.defaultImage}
                           mediumImage={props.mediumImage}
                           highImage={props.highImage}/>
                <Content>
                    <Title>{props.title}</Title>
                    <Paragraph>{props.description}</Paragraph>
                </Content>
            </Link>
        </VideoCardContainer>
    )
}

export default VideoCardComponent;