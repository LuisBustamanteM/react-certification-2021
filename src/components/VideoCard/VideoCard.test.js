import React from 'react';
import {render} from "@testing-library/react";
import VideoCard from './index';
import { BrowserRouter as Router } from 'react-router-dom';
import AppContext from "../../AppContext";
import mockData from '../../MockData/youtubeResult.json'

test("<VideoCard/> matches snapshot", () => {
    let video = mockData.items[0]
    let images = video.snippet.thumbnails

    const component = render(
        <AppContext>
            <Router>
                <VideoCard
                    defaultImage={images.default.url}
                    mediumImage={images.default.medium}
                    highImage={images.default.high}
                    title={video.snippet.title}
                    description={video.snippet.description}/>
            </Router>
        </AppContext>
    );

    expect (component.container).toMatchSnapshot();
});
