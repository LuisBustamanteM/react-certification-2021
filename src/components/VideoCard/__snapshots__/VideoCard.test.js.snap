// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoCard/> matches snapshot 1`] = `
<div>
  <li
    class="sc-bdfBQB bhWnnh"
  >
    <a
      href="/video/undefined"
    >
      <div
        class="sc-gsTEea cpVGqN"
      />
      <div
        class="sc-dlfnuX kyDYvr"
      >
        <h2
          class="sc-hKgJUU kIlOys"
        >
          FIRST IMPRESSIONS WITH SORA IN SMASH ULTIMATE!
        </h2>
        <p
          class="sc-eCstlR dObQUk"
        >
          HERE IT IS EVERYONE! Sora is out in Super Smash Bros Ultimate, and this is my first time with the character. Lets learn him together in a variety of settings, see ...
        </p>
      </div>
    </a>
  </li>
</div>
`;
